### ISSUE TITLE
Request : v2.25 : Random Assign on no winner at chatbidding

### ISSUE THREAD

**Issue by Hysteriana** [Mordonus/RaidOps#289]
see title.. maybe if no winner has been selected by chatbidding, optional the winner could be chosen  randomly ? also optional inside or outside of the usable group.

**Comment by Drutol** [comment]
How about adding third button next to assign and select?

**Comment by Hysteriana** [comment]
yeah sure, why not. btw what happened to select multiple items and then
auto random assign them all piece by piece?
Am 15.07.2015 12:45 schrieb "Piotr Szymczak" notifications@github.com:

**Comment by Drutol** [comment]
Implementation of assignment and ML in general is built on the data from the other addon and not from source, which I deeply regret now but I didn't have a clue what was I doing back then. I'll have to use Delete button on major part of BID.lua and redo this without the CRB's addon (After #285). And down to the point : doing this would be **very** wonky + I've to redo this anyway. The thing you want is already built-in in new ML addon but I cannot get around to do some final things for alpha release.

**Comment by Hysteriana** [comment]
ok, its no biggy :) just 1,2 mouse clicks more :)

**Comment by Drutol** [comment]
Suddenly from very little things to do I've few weeks worth of work yay! ^^

EDIT: Maybe I'll try to push today this ML addon that I postpone and postpone... I'll need some help on the hill. If the concept proves useful then I'll finish it if not I'll abandon it.

**Comment by Drutol** [comment]
Write here if you can go afk on hill :)

**Comment by Drutol** [comment]
https://www.dropbox.com/s/mhopx7cosb633i0/RaidOpsLootDivison.rar?dl=0

**Comment by Renira** [comment]
Chiming in how having a random button would be an incredibly awesome thing. :)

**Comment by Drutol** [comment]
Random button is already here. ^^

New ML addon:
 Random distribution piece by piece is in this new addon (dropbox link). I'm looking for feedback whether the concept is good enough to develop further.( this is somewhat alpha state => not every button works , display timings are a bit off) If you want to check it out just go kill some random mobs with master loot on.

**Comment by Drutol** [comment]
A little how to :
When item drops it will appear in :
- Loot pool
- Random pool
  - If items are left in the random pool they will be distributed at random piece by piece. You can filter what goes to random pool by pressing blue button on the right of this container.
  - Random filters have two options 1. Only put item in random pool 2. Just assign item and forget ( I didn't know that this is already built-in so it's a bit useless)

Each item can be drag&dropped between pools , recipients and action slot ( for bidding - currently not working). When you have finished moving items where needed just press 'Ship items'. There will be small summary.

If you think that this will somehow work then great if not I'll just remove it.

**Comment by Larnen** [comment]
Could we have an option for where it goes if no winner? Either random OR guild bank. We tend to assign things to GB just to eliminate any chance someone may want it but not bid and then get it randomly anyway. Ie its going to be salvaged unless someone bids :)

**Comment by Drutol** [comment]
@Larnen I don't have much time left to play with v2.26. I'll add this in v2.27.

Added v2.26.

Has anyone tried this ML addon?

**Comment by Renira** [comment]
I haven't had a chance yet since I saw it on Monday and I've been raiding all week. Going to make a Dropbox account and try it tonight in a small party. :)

**Comment by Renira** [comment]
I'm really looking forward to a smaller loot window. :)

**Comment by Drutol** [comment]
It can be pretty small , but it may be harder to use. I'm not raiding so I don't know what is cool and what's not. I'll try to think of 'micro' version.

**Comment by Renira** [comment]
I'll let you know!

**Comment by Renira** [comment]
Well, I didn't get to it tonight, but I will this weekend. Sorry and thank you!  Cause this, while awesome...https://www.youtube.com/watch?v=w9MIliTJ0tY (sorry for the terrible quality)...is a little rough to manage, though obviously well worth it. ;)

**Comment by Drutol** [comment]
Them errors... which version is it? Seeing how this works I'll try to make some changes.

You don't have to select this offspec option o pop-up if the value in the box is already 0. There's no offspec flag added , it just decreases the value in the box.

EDIT: Nvm , found the bug.

**Comment by Renira** [comment]
Gotcha and good to know. It was an older version, but I'm glad you found the bug. It was an odd one and at the time I wasn't sure what I had done to trigger it (think it was my mic bind key..and since it was not a current version I wasn't sure it was relevant).
Also, what exactly does the "select" button do in chat bidding? I haven't been able to figure it out. :)
Thanks again!

**Comment by Hysteriana** [comment]
select will select the winner in the master loot window without assigning the item. its a bit redundant in case you want to assign the item your self and not by the chatbidding. you know, there are are settings to skip the epgp popup if assigned by chatbidding. if for some reason you need the popup, you can use selected and masterloot window instead.

**Comment by Renira** [comment]
Hm, odd, I've never been able to get select to work then. I'll play with it more to make sure it's not just exact moments. Can it be used before chat bidding ends (because that's when I've attempted it) or only after? Because if only after, then it's a bit superfluous as I need to select someone before bidding ends for them to appear as the winner in chat. :)
Also yeah, I am aware of the pop-up setting, but I've chosen to keep the pop-up as confirmation it's being counted and logged (I'm a big fan of seeing things) and I use it for quick reference to keep track gp on the side (I just keep notepad notes) in case of a game crash where I lose app data (it's happened and unpleasant).

**Comment by Drutol** [comment]
I've added separate button to assign all selected items randomly:
![image](https://cloud.githubusercontent.com/assets/6102530/8765460/f845739e-2e04-11e5-9e18-624a95d8a391.png)

Requesting hill assistance!

**Comment by Hysteriana** [comment]
run to the hills run for your life..
 Am 19.07.2015 11:58 schrieb "Piotr Szymczak" notifications@github.com:

**Comment by Drutol** [comment]
I'll be online in an hour.

**Comment by Hysteriana** [comment]
ok
Am 19.07.2015 13:07 schrieb "Piotr Szymczak" notifications@github.com:

**Comment by Drutol** [comment]
I'm online

**Comment by Hysteriana** [comment]
that wasnt an hour :) be right there

**Comment by Drutol** [comment]
I feel like closing this issue... @Renira I would be **super** grateful if you made such small showcase video(s) during proper raids after v2.27 is out, so I can put them on curse. Like the chat bidding one... and... tbh I don't know what else ^^
I can organize editing myself , I just need the footage.

Anyway , I'll open separate issue for thsii at a later day.

**Comment by Renira** [comment]
I'll run some footage tonight, but I might not have much for you until Wednesday (progression bang-heads-against-wall night on Mondays). I can also record at a higher resolution, but it'll have a watermark of the program I use. If that's alright, I'll move ahead with that. :)

**Comment by Renira** [comment]
Nevermind, when 2.27 goes live! Helps if I read more closely. I blame work for muddling my brain. :P

**Comment by Hysteriana** [comment]
try using OBS for recording and or broadcasting. its free, fast, resource friendly and very customizable, and no watermarks :) https://obsproject.com/

**Comment by Drutol** [comment]
Win10 has built-in recording feature , but idk how well it works ^^. 

v2,27 because:
- fix to this error ( I'd like to avoid them in the promo material :D)
- this new button seems like a thing that could help you quite a bit and therefore it's worthy of showing

v2.27 beta today

**Comment by Renira** [comment]
Sounds good and I'll check it out Hysteriana.  Thanks for the suggestion! :)  I finally got to testing out the new ML addon.  So far, I'm liking it quite a bit and the fact that it works with NotNowMasterLoot is awesome.  I made some notes while playing with it. :)

1.) Only icons is totally doable, though you may want to make them smaller or offer a text option...not sure how this looks on lower resolutions, but it's still fairly large for me.  The party members look like they can be made smaller as well possibly (class icon on the left can be shrunk probably).
2.) Chat bidding not working. (I have a feeling this is one of those buttons not active yet. :smile: )
3.) Can't hide the base master loot frame.  I'm not sure if this is even doable with the way Carbine has everything set up.  I can resize it to teeny tiny and just attempt to ignore it. :P 
4.) Can't close master loot window once it's opened.  If a fight starts back up again, it might be nice to close the window so it's hidden again by NotNowMasterLoot or so it minimizes in some way.  I clicked the red close button, but got nothing from it.
5.) Tooltips in the summary window.  Something simple by the checkmarks like "assign" and something for the big button at the bottom "assign all and close" or some such.  Not necessary, but nicer for user friendliness.
6.) I'm not sure if GP amount is currently being factored by settings in RaidOps.  I don't have small loot adjusted in the settings so I need to test this with higher level loot drops.  It's definitely calculating something, so it's probably fine. :P
![renira_spiffydiddle 150720 222013](https://cloud.githubusercontent.com/assets/13113326/8793778/39b7b450-2f30-11e5-91c7-a7263036cd6f.png)
8.) Popped error when trying to assign loot through summary by clicking checkmark next to name:
...tar\Addons\RaidOpsLootDivison
\RaidOpsLootDivison.lua:1423: attempt to index global 'child' (a nil value)
stack trace:    ...tar\Addons
\RaidOpsLootDivison\RaidOpsLootDivison.lua:1423: in function <...tar\Addons\RaidOpsLootDivison
\RaidOpsLootDivison.lua:1421>
![renira_spiffydiddle 150720 222231](https://cloud.githubusercontent.com/assets/13113326/8793744/06ff8380-2f30-11e5-84b2-79952c292621.png)
9.) The settings window could use a little more explanation. :)  I just kept clicking some things off and on to see what they did. ;)  I assume most of that would be explained in the wiki.

Most of the app seems to be working great. :)  The drag/drop is functioning as intended. The random is distributing to multiple players. No errors when moving items around within the app.  Overall, I like it.  I just wish I could get rid of that default. ;)

**Comment by Drutol** [comment]
Wow thanks for this summary!
- Hmm , I'll add an option to create list of items with names, hide random pool and display check-boxes for random items.
- Yup integration with main addon isn't currently implemented.
- I left this on in order to verify whether loot isn't duplicating / missing (I had such problems in the past)
- This is the button that misses his functionality somehow.
- Ok.
- Tooltips are processed globally . I'm just stealing CRB's function , wrapping it in my own stuff , and returning it like nothing has happened.
- Got it.
- It's the WIP version , to check whether somebody would like it. I'll now resume work on this thing.

Once again , thanks for the input :D. 

What do you mean by "it works with NotNowMasterLoot"? I know what is this thing , but it shouldn't work with my addon , or you just mean this button in the upper left corner?

**Comment by Renira** [comment]
Yeah, I thought it worked together, but that is totally your addon popping the button and not the other addon. Been so long since I'd seen it that I forgot it looked different. Sorry about that. ;)

**Comment by Hysteriana** [comment]
just an idea, instead of all this buttons, maybe just one button that opens a pulldown menu ?
and all icon buttons turned into text buttons inside that pulldown menu ? space problem solved :)

**Comment by Drutol** [comment]
For chatbidding's assign,select and random?

**Comment by Hysteriana** [comment]
no i mean on the main page, like you already did next to he the selected member counter.
you could put the other buttons there too

**Comment by Drutol** [comment]
There would be just a big hole of empty space then. Right now it looks pretty ok imo , and the buttons used most often are easy to access.
